ksp_plugin_adapter/main_window.cs

```diff
@@ -0,0 +1,504 @@
+          Convert.ToBoolean(show_prediction_settings_value);
+    }
+
+    String history_length_index_value =
+        node.GetAtMostOneValue("history_length_index");
+    if (history_length_index_value != null) {
+      should_load_compatibility_data_ = false;
+      history_length_index_ = Convert.ToInt32(history_length_index_value);
+    }
+
+    String buffered_logging_value =
+        node.GetAtMostOneValue("buffered_logging");
+    if (buffered_logging_value != null) {
+      buffered_logging_ = Convert.ToInt32(buffered_logging_value);
+    }
+    String stderr_logging_value = node.GetAtMostOneValue("stderr_logging");
+    if (stderr_logging_value != null) {
+      stderr_logging_ = Convert.ToInt32(stderr_logging_value);
+    }
+    String suppressed_logging_value =
+        node.GetAtMostOneValue("suppressed_logging");
+    if (suppressed_logging_value != null) {
+      suppressed_logging_ = Convert.ToInt32(suppressed_logging_value);
+    }
+    String verbose_logging_value = node.GetAtMostOneValue("verbose_logging");
+    if (verbose_logging_value != null) {
+      verbose_logging_ = Convert.ToInt32(verbose_logging_value);
+    }
+
+    String must_record_journal_value =
+        node.GetAtMostOneValue("must_record_journal");
+    if (must_record_journal_value != null) {
+      must_record_journal_ = Convert.ToBoolean(must_record_journal_value);
+    }
+
+    Log.SetBufferedLogging(buffered_logging_);
+    Log.SetSuppressedLogging(suppressed_logging_);
+    Log.SetStderrLogging(stderr_logging_);
+    Log.SetVerboseLogging(verbose_logging_);
+
+    if (must_record_journal_) {
+      journaling_ = true;
+      Log.ActivateRecorder(true);
+    }
+  }
+
+  public override void Save(ConfigNode node) {
+    base.Save(node);
+
+    node.SetValue("show_ksp_features",
+                  show_ksp_features_,
+                  createIfNotFound : true);
+    node.SetValue("show_logging_settings",
+                  show_logging_settings_,
+                  createIfNotFound : true);
+    node.SetValue("show_prediction_settings",
+                  show_prediction_settings_,
+                  createIfNotFound : true);
+
+    node.SetValue("history_length_index",
+                  history_length_index_,
+                  createIfNotFound : true);
+
+    node.SetValue("buffered_logging",
+                  buffered_logging_,
+                  createIfNotFound : true);
+    node.SetValue("stderr_logging",
+                  stderr_logging_,
+                  createIfNotFound : true);
+    node.SetValue("suppressed_logging",
+                  suppressed_logging_,
+                  createIfNotFound : true);
+    node.SetValue("verbose_logging",
+                  verbose_logging_,
+                  createIfNotFound : true);
+
+    node.SetValue("must_record_journal",
+                  must_record_journal_,
+                  createIfNotFound : true);
+  }
+
+  protected override string Title { get; } = "Principia";
```

Comment:
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/properties#expression-body-definitions
```C#
protected override string Title => "Principia";
```